src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -81,6 +81,44 @@ struct kds_cu_info
   }
 };
 
+/**
+ * qspi write protection status
+ * byte 0:
+ *   0: status not available, 1: status available
+ * byte 1 primary qspi(if status available):
+ *   1: write protect enable, 2: write protect disable
+ * byte 2 recovery qspi(if status available):
+ *   1: write protect enable, 2: write protect disable
+ */
+struct qspi_status
+{
+  using result_type = query::xmc_qspi_status::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    auto pdev = get_pcidev(device);
+  
+    std::string status_str, errmsg;
+    pdev->sysfs_get("xmc", "xmc_qspi_status", errmsg, status_str);
+    if (!errmsg.empty())
+      throw std::runtime_error(errmsg);
+
+    std::string primary, recovery;
+    for (auto x : status_str) {
+      int status_byte = x - '0';
```

Comment:
What happens if 'x' is less then '0' (48) or greater then '9' (57).  

Wouldn't it be better if this code simply use the actual characters for comparison?  For example:

```
 if(status_byte == '0')
        return std::pair<std::string, std::string>("N/A", "N/A");

      if(primary.empty())
        primary = status_byte == '1' ? "Enabled": status_byte == '2' ? "Disabled" : "Invalid";
      else
        recovery = status_byte == '1' ? "Enabled": status_byte == '2' ? "Disabled" : "Invalid"
```

Reply:
Ah yes. This is so much better! Thank you